src/pages/payments/edit/PaymentFields.tsx

```diff
@@ -9,61 +9,86 @@
  */
 
 import { Card, Element } from '@invoiceninja/cards';
-import { endpoint } from 'common/helpers';
+import { enterprisePlan } from 'common/guards/guards/enterprise-plan';
+import { proPlan } from 'common/guards/guards/pro-plan';
+import { endpoint, isHosted } from 'common/helpers';
 import { request } from 'common/helpers/request';
+import { toast } from 'common/helpers/toast/toast';
 import { useHandleCustomFieldChange } from 'common/hooks/useHandleCustomFieldChange';
 import { useInjectCompanyChanges } from 'common/hooks/useInjectCompanyChanges';
 import { updateRecord } from 'common/stores/slices/company-users';
+import { CustomFieldsPlanAlert } from 'components/CustomFieldsPlanAlert';
 import { Field } from 'pages/settings/custom-fields/components';
-import toast from 'react-hot-toast';
 import { useTranslation } from 'react-i18next';
 import { useDispatch } from 'react-redux';
 
 export function PaymentFields() {
-  const company = useInjectCompanyChanges();
+  const [t] = useTranslation();
+
   const dispatch = useDispatch();
 
-  const [t] = useTranslation();
+  const company = useInjectCompanyChanges();
+
   const handleCustomFieldChange = useHandleCustomFieldChange();
 
+  const disabledCustomFields = !proPlan() && !enterprisePlan() && isHosted();
+
   const onSave = () => {
-    const toastId = toast.loading(t('processing'));
+    toast.processing();
 
     request(
       'PUT',
       endpoint('/api/v1/companies/:id', { id: company?.id }),
       company
-    ).then((response) => {
-      dispatch(updateRecord({ object: 'company', data: response?.data.data }));
+    )
+      .then((response) => {
+        dispatch(
+          updateRecord({ object: 'company', data: response?.data.data })
+        );
 
-      toast.success(t('updated_payment'), { id: toastId });
-    });
+        toast.success('updated_payment');
+      })
+      .catch((error) => {
+        console.error(error);
+        toast.error();
+      });
   };
 
   return (
-    <Card title={t('custom_fields')} withSaveButton onFormSubmit={onSave}>
-      <Element
-        leftSide={
-          <div className="inline-flex items-center space-x-2">
-            <span>{t('note')}</span>
-            <span className="text-red-600">*</span>
-          </div>
-        }
+    <>
+      <CustomFieldsPlanAlert />
+
+      <Card
+        title={t('custom_fields')}
+        withSaveButton
+        onFormSubmit={onSave}
+        disableSubmitButton={disabledCustomFields}
+        disableWithoutIcon={disabledCustomFields}
       >
-        Custom fields apply to all Payments, they are not specific to this one.
-        <i>Needs translation.</i>
-      </Element>
+        <Element
+          leftSide={
+            <div className="inline-flex items-center space-x-2">
+              <span>{t('note')}</span>
+              <span className="text-red-600">*</span>
+            </div>
+          }
+        >
+          Custom fields apply to all Payments, they are not specific to this
```

Comment:
Can we translate this string?

Reply:
I discussed with David on the Slack for this one, and he suggested me to completely remove this. So it's removed.